omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -586,11 +581,30 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
                 else
                     self._table.numberOfPages = Math.floor(initialRowCount/self._table.fnSettings()._iDisplayLength)+1;
 
-            } else if(self.options.searchPhrase || self.options.doSearchWhenEmpty) {
-                if (self.options.searchPhrase == null) {
-                    self.options.searchPhrase = "";
+            } else{ 
+            	if(self.options.searchPhrase || self.options.doSearchWhenEmpty) {
+            		if (self.options.searchPhrase == null) {
+                        self.options.searchPhrase = "";
+                    }                
+                    //check to see if we need to select the checkbox for includeVerbose and/or includeVoided
+                    //since there is a searchPhrasese the page will reload using these values
+                	if(self.options.showVerbose){
+                		$j('#includeVerbose').attr('checked','checked');
+                	}
+                	if(self.options.includeVoided){
+                		$j('#includeVoided').attr('checked','checked');
+                	}
+                	$j(input).val(self.options.searchPhrase).keyup();
+                	//setting the cursor to the end of the searchPhrase
+                    //setSelectionRange is a method of the DOM element not the jQuery object so get the DOM element
+                    var inputNodeDOM = $j('#inputNode').get(0);
+                    if(inputNodeDOM){
+                    	var searchPhraseLength = self.options.searchPhrase.length;
+                    	if (inputNodeDOM.setSelectionRange) {
+                    		inputNodeDOM.setSelectionRange(searchPhraseLength, searchPhraseLength);
```

Comment:
This is where the search gets triggered when the search box gets populated by passed in data rather than the user typing in data to trigger a keyup. Is there something I am not seeing that should make that happen somewhere else?